compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/symtab_transformer/gen_c_transformer/nondet_transformer.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use super::super::super::{Expr, Location, Stmt, Symbol, SymbolTable, Type};
+use super::super::Transformer;
+use super::common::type_to_string;
+use rustc_data_structures::fx::FxHashMap;
+
+/// Struct for handling the nondet transformations for --gen-c-runnable.
+pub struct NondetTransformer {
+    new_symbol_table: SymbolTable,
+    nondet_types: FxHashMap<String, Type>,
+}
+
+impl NondetTransformer {
+    /// Transform all identifiers in the symbol table to be valid C identifiers;
+    /// perform other clean-up operations to make valid C code.
+    pub fn transform(original_symbol_table: &SymbolTable) -> SymbolTable {
+        let new_symbol_table = SymbolTable::new(original_symbol_table.machine_model().clone());
+        NondetTransformer { new_symbol_table, nondet_types: FxHashMap::default() }
+            .transform_symbol_table(original_symbol_table)
+    }
+
+    /// Extract `nondet_types` map for final processing.
+    pub fn nondet_types_owned(&mut self) -> FxHashMap<String, Type> {
+        std::mem::replace(&mut self.nondet_types, FxHashMap::default())
+    }
+}
+
+impl Transformer for NondetTransformer {
+    /// Get reference to symbol table.
+    fn symbol_table(&self) -> &SymbolTable {
+        &self.new_symbol_table
+    }
+
+    /// Get mutable reference to symbol table.
+    fn mut_symbol_table(&mut self) -> &mut SymbolTable {
+        &mut self.new_symbol_table
+    }
+
+    /// Get owned symbol table.
+    fn extract_symbol_table(self) -> SymbolTable {
+        self.new_symbol_table
+    }
+
+    /// Transform nondets to create default values for the expected type.
+    fn transform_expr_nondet(&mut self, typ: &Type) -> Expr {
+        let transformed_typ = self.transform_type(typ);
+
+        let identifier = format!("non_det_{}", type_to_string(&transformed_typ));
+        let function_type = Type::code(vec![], transformed_typ);
+
+        // Create non_det function which returns default value in postprocessing
+        self.nondet_types.insert(identifier.clone(), function_type.clone());
+
+        Expr::symbol_expression(identifier, function_type).call(vec![])
+    }
+
+    /// Don't transform padding fields so that they are ignored by CBMC --dump-c.
```

Comment:
Give example of what happens if you do and don't do this

Reply:
Done.